File: practice/algos/rect_functions/rectangle.rb
class Rectangle
  attr_reader :width, :height

  def initialize(args = { })
    @width  = args.fetch(:width, 1)
    @height = args.fetch(:height, 1)
  end

  def area
    width * height
  end

  def perimeter
    (height * 2) + (width * 2)
  end

  def diagonal
    Math.sqrt((height^2) + (width^2))
  end

  def is_square?
    return true if rectangle.height == rectangle.width
  	  false
  end

end

rectangle_one = Rectangle.new(height: 4, width: 5)

puts rectangle_one.area
puts rectangle_one.perimeter
puts rectangle_one.diagonal
puts rectangle_one.is_square?

File: practice/algos/rect_functions/rectangle.js
var rectangle1 = {width: 4, height: 4}


function area(rectangle) {
	return rectangle.width * rectangle.height;
}

var areaOfRectangle1 = area(rectangle1);

console.log(areaOfRectangle1)

function perimeter(rectangle) {
  return (2 * rectangle.height) + (2 * rectangle.width);
}

function diagonal(rectangle) {
  return Math.sqrt((width * width) + (height * height))
  // return Math.sqrt(Math.pow(width, 2) + Math.pow(height, 2))
}

function isSquare(rectangle) {
  if (rectangle.height === rectangle.width) {
    return true;
  } else {
    return false;
  }
}
